go/staking/tests/tester.go

```diff
@@ -48,86 +89,58 @@ func StakingImplementationTests(
 	entitySigner signature.Signer,
 	runtimeID common.Namespace,
 ) {
+	state := newStakingTestsState(t, backend, consensus)
```

Comment:
(here and below) shouldn't this be on the inside of the loop? or do tests mutate this?

Reply:
Currently, tests don't mutate state but yes, perhaps it would be even cleaner to create a new state for each test that is run.

Reply:
but the tests transfer things around and then nothing calls `update`, so wouldn't that go wrong?

Reply:
> but the tests transfer things around and then nothing calls `update`, so wouldn't that go wrong?

Yea, tests move things around but they currently handle account updates manually, e.g.:
https://github.com/oasislabs/oasis-core/blob/5435fade6371883b3dc215cd7fec8737b91997a9/go/staking/tests/tester.go#L355-L358